### ISSUE TITLE
[Footer] Email signup relies on native validation

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#399]
## Issue summary

When the email signup form is submitted with invalid data, the error message is displayed using platform native validation.

Native validation is problematic as:

- The error message is not conveyed in a manner that is easily correctable or memorable (non-specific messaging)
- The message may not remain visible long enough for sighted users with a cognitive impairment to consume
- Low-vision zoom users may miss the notification completely

<details>
  <summary>Screenshots</summary>

<img width="404" alt="Dawn theme footer section. Error message is displayed in a Chrome native notification bubble." src="https://user-images.githubusercontent.com/1392632/129241783-af3a85bb-657c-4463-b8ab-f108ae36d103.png">

</details>

#### HTML

```html
<input
  id="NewsletterForm--footer"
  type="email"
  name="contact[email]"
  class="field__input"
  value=""
  aria-required="true"
  autocorrect="off"
  autocapitalize="off"
  autocomplete="email"
  placeholder="Email"
  required=""
/>
```

## Steps to reproduce

1. Go to a Dawn homepage which features the Emal signup section with any supported browser
2. Submit the form with invalid data

### Expected

- Custom error text to be placed near the related `input`, attached via `aria-describedby` attribute.

### Actual

- `input` relies on native validation.

## Recommendation

1. Remove the `required` attribute to remove native validation.
2. When an error occurs, include an error message next to the related `input` with a unique `id` value.
3. When an error occurs, add the `aria-describedby` attribute to the `input`, setting its value to the `id` of the error container.
4. When an error occurs, shift keyboard focus back to the `input`.

#### HTML

```html
<input
  id="NewsletterForm--footer"
  type="email"
  name="contact[email]"
  class="field__input"
  value=""
  aria-required="true"
  autocorrect="off"
  autocapitalize="off"
  autocomplete="email"
  placeholder="Email"
  aria-describedby="emailSignupError"
/>
<small class="error" id="emailSignupError">
  <svg><!-- Error icon… --></svg>
  Please enter a valid email address.
</small>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **WCAG Principal:** Understandable
- **WCAG SC:** [3.3.1 Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html)
- **Severity:** High
- **Effort:** Medium
- **Related:**: https://github.com/Shopify/dawn/issues/700

**Comment by bakura10** [comment]
Should not this issue rather be sent to browsers bug tracker ? If such an important feature as input validation fail to provide accessibility, i feel it would be better to work with browser vendors to improve that.

**Comment by bakura10** [comment]
Browser vendors are pretty keen to fix accessibility issues these days so it definitely seems to be a better way and will improve the web for everybody.

@svinkle , can you please issue a report for browser vendor with the results of your research?

Blink bug tracker (Chrome): https://bugs.chromium.org/p/chromium/issues/list
WebKit bug tracker (Safari): https://bugs.webkit.org/
Gecko bug tracker (Firefox): https://bugzilla.mozilla.org/describecomponents.cgi

I would have loved to do it to help but I don't have a knowledge as good as yours when it comes to this.
